squad/core/management/commands/fix_squadplugin_data.py

```diff
@@ -21,31 +21,33 @@ def handle(self, *args, **options):
         show_progress = options['show_progress']
         batch_size = options['batch_size']
         android_suites = ['cts', 'vts']
-        condition = Q(suite__slug__icontains=android_suites[0])
-        condition |= Q(suite__slug__icontains=android_suites[1])
-        logger.info("fetching %s tests" % (batch_size if batch_size else 'all'))
-        tests = Test.objects.filter(condition).exclude(metadata__suite=F('suite__slug')).prefetch_related('suite', 'metadata').defer('log')
+        condition = Q(suite__icontains=android_suites[0])
+        condition |= Q(suite__icontains=android_suites[1])
+        logger.info("fetching %s metadata for cts and vts tests" % (batch_size if batch_size else 'all'))
+        cts_and_vts_metadata = SuiteMetadata.objects.filter(condition, kind='test')
         if batch_size:
             tests = tests[:batch_size]
         count_processed = 0
-
         buggy_metadata = set()
-
-        for test in tests:
-            if test.metadata.suite != test.suite.slug:
-                metadata, _ = SuiteMetadata.objects.get_or_create(kind='test', suite=test.suite.slug, name=test.name)
-                buggy_metadata.add(metadata)
-                test.metadata = metadata
-                test.save()
-            count_processed += 1
-            if count_processed % 1000 == 0 and show_progress:
-                print('.', end='', flush=True)
-
-        deleted_metadata_count = 0
-        for metadata in buggy_metadata:
-            test_count = Test.objects.filter(metadata=metadata).count()
-            if test_count == 0:
-                metadata.delete()
-                deleted_metadata_count += 1
-
-        print('Deleted %d buggy metadata objects' % deleted_metadata_count)
+        for s in cts_and_vts_metadata: 
+             print("Processing SuiteMetedata with suite %s" % s.suite, flush=True) 
+             first_test = Test.objects.filter(metadata=s).first()      
+             if first_test == None: 
+                 print("Deleting Suitemetadata with suite %s" % s.suite, flush=True) 
+                 s.delete() 
+                 continue 
+             print("Found Test with for above Suitemetadata. Test.suite is %s " % first_test.suite, flush=True) 
+             correct_suite = first_test.suite.slug 
+             if correct_suite != s.suite:  
+                 correct_metadata = SuiteMetadata.objects.filter(kind='test', suite=correct_suite).first() 
+                 if correct_metadata:
+                     print("Found existing SuiteMetadata with correct suite %s. Will use it to update all affected tests" % correct_metadata, flush=True) 
+                     print("correct_metadata: %s" % correct_metadata.id, flush=True) 
+                     print("Updating tests", flush=True)             
+                     all_affected_tests= Test.objects.filter(metadata=s) 
+                     all_affected_tests.update(metadata=correct_metadata)              
```

Comment:
`update` returns the number of updated objects, it would be nice to print how many objects got updated as well